roles/download/tasks/prep_kubeadm_images.yml

```diff
@@ -7,9 +7,13 @@
     - not skip_downloads | default(false)
     - downloads.kubeadm.enabled
 
+- name: Prep_kubeadm_images | Set kubeadm api version based on kubernetes version
+  set_fact:
+    kubeadmConfig_api_version: "{{ 'v1beta4' if kube_version is version('v1.31.0', '>=') else 'v1beta3' }}"
```

Comment:
This should be in a `defaults` or `vars` (depending if there is a good reason for user to override this, which I don't think there is) file rather than a fact.
Given it's used at multiples places, kubespray-defaults seems a good candidate.

Reply:
Yes, this variable will rename to `kubeadm_config_api_version` and move to `roles/kubespray-defaults`.